agent/proxycfg/state.go

```diff
@@ -432,6 +437,54 @@ func (s *state) initWatchesMeshGateway() error {
 	return err
 }
 
+func (s *state) initWatchesIngressGateway() error {
+	// Watch for root changes
+	err := s.cache.Notify(s.ctx, cachetype.ConnectCARootName, &structs.DCSpecificRequest{
+		Datacenter:   s.source.Datacenter,
+		QueryOptions: structs.QueryOptions{Token: s.token},
+		Source:       *s.source,
+	}, rootsWatchID, s.ch)
+	if err != nil {
+		return err
+	}
+
+	// Watch the leaf cert
+	err = s.cache.Notify(s.ctx, cachetype.ConnectCALeafName, &cachetype.ConnectCALeafRequest{
+		Datacenter:     s.source.Datacenter,
+		Token:          s.token,
+		Service:        s.service,
+		EnterpriseMeta: s.proxyID.EnterpriseMeta,
+	}, leafWatchID, s.ch)
+	if err != nil {
+		return err
+	}
+
+	// Watch the service's own ingress-gateway config entry
+	err = s.cache.Notify(s.ctx, cachetype.ConfigEntryName, &structs.ConfigEntryQuery{
```

Comment:
Noting that this means the ingress gateway will need to be given a token with `operator:read` per your definition of `CanRead()` on this config entry type.

Reply:
I think that's what we want (we've been thinking operators would be the ones setting up ingress gateways) but it's worth calling out in case there's a reason to use a different acl resource type (or create a new one).

Reply:
After more thinking on this, it feels weird to require something that has read privileges for the miscellaneous operator things that we've tied to `operator:read`. This should probably require something else.

Reply:
We updated the `CanRead` function to only require `service:read` on the gateway itself.